compiler/test/fail_compilation/templatethis.d

```diff
@@ -0,0 +1,52 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/templatethis.d(11): Error: cannot use `this` outside an aggregate type
+fail_compilation/templatethis.d(15): Error: cannot use `this` outside an aggregate type
+fail_compilation/templatethis.d(19): Error: cannot use `this` outside an aggregate type
+fail_compilation/templatethis.d(21): Error: cannot use `this` outside an aggregate type
+---
+*/
+
+template t(this T)
+{
+}
+
+struct S(this T)
+{
+}
+
+enum e(this T) = 1;
+
+void f(this T)()
+{
+}
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/templatethis.d(34): Error: cannot use `this` outside an aggregate type
+fail_compilation/templatethis.d(37): Error: mixin `templatethis.t2!()` error instantiating
+---
```

Comment:
I'm not a fan of breaking up the TEST_OUTPUT directives, if the diagnostic were to change in the future, then it can't be updated in an automated way.

Reply:
(cannot be updated... yet)

Reply:
Note: There are 85 files in `fail_compilation` which already do this according to a quick grep.

Reply:
And they are annoying to deal with as one must manually work out where to update them when diagnostics change.

Reply:
Yes, I agree that it's better to just group all the error messages into a single `TEST_OUTPUT` directive.

Reply:
@ibuclaw OK, I see. Fixed!